cpp/memilio/math/integrator.h

```diff
@@ -43,13 +40,25 @@ class IntegratorCore
     virtual ~IntegratorCore(){};
 
     /**
-     * @brief Step of the integration with possible adaptive with
+     * @brief Make a single integration step.
      *
-     * @param[in] f Right hand side of ODE
-     * @param[in] yt value of y at t, y(t)
-     * @param[in,out] t current time step h=dt
-     * @param[in,out] dt current time step h=dt
-     * @param[out] ytp1 approximated value y(t+1)
+     * The behaviour of this method changes slightly when the integration scheme has adaptive step sizing. 
+     * These changes are noted in (brackets) below.
+     * Adaptive integrators must have bounds dt_min and dt_max for dt.
+     * The adaptive step sizing is considered to have failed, if dt would be adapted to a value strictly below dt_min.
+     * Even if the step sizing failed, the integrator must make a step of at least size dt_min.
+     *
+     * @param[in] f Right hand side of ODE. May be called multiple times with different arguments.
+     * @param[in] yt The known value of y at time t.
+     * @param[in,out] t The current time. It will be increased by dt.
+     *     (If adaptive, the increment is from [dt_min, dt].)
+     * @param[in,out] dt The current step size h=dt.
```

Comment:
Is it the current or the next step size? Below it is said that dt is an estimate of the next step, so it should be the next step size, rights?

Reply:
I'll need rephrase this - dt [in] is the current step size, dt [out] is _set to_ the next. The latter does not happen with non-adaptive steppers.